ambry-network/src/test/java/com.github.ambry.network/SelectorTest.java

```diff
@@ -94,6 +92,22 @@ public void testClientDisconnect()
   }
 
   /**
+   * Validate that the client's connectionId is returned via disconnected list after closing
+   */
+  @Test
+  public void testClientClose()
+      throws Exception {
+    String connectionId = blockingConnect();
+    selector.close(connectionId);
+    selector.poll(10);
+    assertEquals("Request should not have succeeded", 0, selector.completedSends().size());
+    assertEquals("There should be a disconnect", 1, selector.disconnected().size());
+    assertTrue("The disconnect should be from our node", selector.disconnected().contains(connectionId));
+    connectionId = blockingConnect();
```

Comment:
I don't think it is useful - it seems to me like a half hearted check and not really of relevance. What errors are you hoping to catch? Isn't the code unrelated?